pep_sphinx_extensions/pep_processor/transforms/pep_contents.py

```diff
@@ -45,19 +44,37 @@ def __init__(self, document, startnode=None):
         self.backlinks = None
 
     def apply(self) -> None:
-        # used in parts.Contents.build_contents
-        self.toc_id = self.startnode.parent["ids"][0]
-        self.backlinks = self.document.settings.toc_backlinks
-
-        # let the writer (or output software) build the contents list?
-        if getattr(self.document.settings, "use_latex_toc", False):
-            # move customisation settings to the parent node
-            self.startnode.parent.attributes.update(self.startnode.details)
-            self.startnode.parent.remove(self.startnode)
+        contents = self.build_contents(self.document[0][4:])  # skip PEP title, headers, <hr/>, and contents
+        if contents:
+            self.startnode.replace_self(contents)
         else:
-            contents = self.build_contents(self.document[0])
-            if contents:
-                self.startnode.replace_self(contents)
-            else:
-                # if no contents, remove the empty placeholder
-                self.startnode.parent.parent.remove(self.startnode.parent)
+            # if no contents, remove the empty placeholder
+            self.startnode.parent.parent.remove(self.startnode.parent)
+
+    def build_contents(self, node, level=0):
+        level += 1
+        entries = []
+        try:
+            children = node.children
+        except AttributeError:
+            children = node
```

Comment:
```suggestion
        children = getattr(node.children, node)
```

Reply:
I've always found getattr a bit magic and avoided ;-)

Reply:
done